uvicorn/main.py

```diff
@@ -359,10 +358,10 @@ def main(
         "use_colors": use_colors,
         "factory": factory,
     }
-    run(**kwargs)
+    run(app, **kwargs)
 
 
-def run(app, **kwargs):
+def run(app: str, **kwargs: typing.Any) -> None:
```

Comment:
https://github.com/encode/uvicorn/pull/990#issuecomment-794321115

> I'm not sure if the app parameter on `run()` should be str? 🤔 On the main I think it is, but on `run`, should it? On `Config`, I'm sure it's not.

As mentioned in the related PR encode/uvicorn#992, `app: Union[ASGIApplication, str]` could work, based on the definition added to `uvicorn._types`. If not, a less strict annotation like `Union[Callable, str]` could work.

It could be helpful to be slightly more specific with the annotation of `kwargs` also. The `kwargs` should be a `dict`, so `**kwargs: dict`.

Reply:
> It could be helpful to be slightly more specific with the annotation of kwargs also. The kwargs should be a dict, so **kwargs: dict.

https://www.python.org/dev/peps/pep-0484/#arbitrary-argument-lists-and-default-argument-values

Reply:
Sorry for the confusion there, I was tired and thinking of the packed forms. I meant to put the `dict` annotation on the kwargs object under `def main`, because you would need an annotation there.

To use something more specific than `Any`, something like this could work, based on the annotations you added:

```py
def run(
    app: str, **kwargs: typing.Union[typing.Optional[list], bool, float, int, str]
) -> None:
    config = Config(app, **kwargs)
    server = Server(config=config)
    ...

```

It's a bit clumsy to list off all those types, so a [type alias](https://mypy.readthedocs.io/en/stable/kinds_of_types.html#type-aliases) could be helpful.

Reply:
I've applied the `app` one, thanks.

As for the `**kwargs`... I'm not sure.